src/styles/transformers/rtl.js

```diff
@@ -0,0 +1,78 @@
+const reTranslate = /((^|\s)translate(3d|X)?\()(\-?[\d]+)/;
+const reSkew = /((^|\s)skew(x|y)?\()\s*(\-?[\d]+)(deg|rad|grad)(,\s*(\-?[\d]+)(deg|rad|grad))?/;
+
+/**
+ * This function ensures that `style` supports both ltr and rtl directions by
+ * checking `styleConstants` in `muiTheme` and replacing attribute keys if
+ * necessary.
+ */
+export default (muiTheme) => (style) => {
+  // Left to right is the default. No need to flip anything.
+  if (!muiTheme || !muiTheme.isRtl) return style;
```

Comment:
> the auto-prefix transformer is also technically checking muiTheme for on/off behavior

I was going to say that it's true but different. We do it only once in the auto prefixer.
I like your proposal.